src/presentation/test/integration/registration/generalPartner/address/postcode-general-partner-principal-office-address.test.ts

```diff
@@ -0,0 +1,96 @@
+import request from "supertest";
+
+import enTranslationText from "../../../../../../../locales/en/translations.json";
+import cyTranslationText from "../../../../../../../locales/cy/translations.json";
+
+import { appDevDependencies } from "../../../../../../config/dev-dependencies";
+import app from "../../../app";
+
+import { getUrl, setLocalesEnabled, toEscapedHtml, testTranslations } from "../../../../utils";
+import { POSTCODE_GENERAL_PARTNER_PRINCIPAL_OFFICE_ADDRESS_URL } from "presentation/controller/addressLookUp/url";
+import GeneralPartnerBuilder, {
+  generalPartnerLegalEntity
+} from "../../../../builder/GeneralPartnerBuilder";
+import AddressPageType from "../../../../../controller/addressLookUp/PageType";
+
+describe("Postcode general partner's principal office address page", () => {
+  const URL = getUrl(POSTCODE_GENERAL_PARTNER_PRINCIPAL_OFFICE_ADDRESS_URL);
+
+  beforeEach(() => {
+    setLocalesEnabled(false);
+    appDevDependencies.cacheRepository.feedCache(null);
+    appDevDependencies.generalPartnerGateway.feedGeneralPartners([]);
+
+  });
+
+  describe("Get postcode general partner's principal office address page", () => {
+    it("should load the principal office address page with English text", async () => {
+      setLocalesEnabled(true);
+
+      const generalPartner = new GeneralPartnerBuilder()
+        .withId(appDevDependencies.generalPartnerGateway.generalPartnerId)
+        .isLegalEntity()
+        .build();
+
+      appDevDependencies.generalPartnerGateway.feedGeneralPartners([generalPartner]);
+
+      const res = await request(app).get(URL + "?lang=en");
+
+      expect(res.status).toBe(200);
+      expect(res.text).toContain(toEscapedHtml(enTranslationText.address.findPostcode.principalOfficeAddress.whatIsPrincipalOfficeAddress) + ` - ${enTranslationText.service} - GOV.UK`);
+      testTranslations(res.text, enTranslationText.address.findPostcode, [
+        "registeredOfficeAddress",
+        "principalPlaceOfBusiness",
+        "usualResidentialAddress",
+        "errorMessages"
+      ]);
+      expect(res.text).not.toContain("WELSH -");
+      expect(res.text).toContain(generalPartner.data?.forename?.toUpperCase());
```

Comment:
the page shouldn't contain the general partner forename and surname but these tests are probably passing as generalPartner.data?.forename and generalPartner.data?.surname are probably "" or null.
what it should do is check that res.text does not contain generalPartnerPerson.forename and generalPartnerPerson.surname.
you can import generalPartnerPerson from the same file as the GeneralPartnerBuilder - src/presentation/test/builder/GeneralPartnerBuilder.ts

Reply:
Ensure checks are correct